add warning for datasets with overlapping datapoints

diegodoimo opened the pull request:
added explicit warnings (and relative pytest files) to the distance matrices computed in  'compute_nn_distances' and 'return_id_gride' when distances of nearby points is close to zero (<=1.1 eps). This warning does not stop the normal workflow.

imacocco commented:
Curious there are tests failing in different versions of python.. I think it's the usual problem of rounding. I'd say to solve it by substituting assert( (a==b).all() ) with assert np.allclose(a,b)

diegodoimo commented:
I have solved that thing though I didn't add/modify anything in  the test_interpolators  file. 

I added a new test_utils folder. Flake8 complains that there is no docstring to some public function related to the tests I added. The message is clear though I didn't understand it fully (e.g. which public functions is it talking about?). For me the commit is clean. Aldo can you understand the issue about flake8?

AldoGl commented:
Hi @diegodoimo, I checked the flake8 errors and they seem quite clear. 

I read:

tests/test_id_estimation/test_gride.py:50:1: D103 Missing docstring in public function
tests/test_utils/test_distances_utils.py:1:1: D100 Missing docstring in public module
tests/test_utils/test_distances_utils.py:7:1: D103 Missing docstring in public function

Essentially you a docstring is needed for the new module "test_distances_utils.py" and for the two functions named both "test_zero_dist" in "test_gride.py" and in "test_distances_utils.py".

Let me know if you want to make these changes or I can do it!

AldoGl commented:
Anyway I like this new feature, and in my opinion it can be merged when we solve the linting problems

AldoGl commented on tests/test_utils/test_distances_utils.py:
```diff
@@ -0,0 +1,10 @@
+import numpy as np
```
A small comment. At the beginning of this module, in addition to the module docstring, we could add the licence. Just like in the other modules, see https://github.com/sissa-data-science/DADApy/blob/main/tests/test_id_estimation/test_2NN.py

AldoGl commented:
✌🏻
